src/linker/Linker.Dataflow/FlowAnnotations.cs

```diff
@@ -373,15 +374,11 @@ bool ScanMethodBodyForFieldAccess (MethodBody body, bool write, out FieldDefinit
 			return true;
 		}
 
-		static bool IsTypeInterestingForDataflow (TypeReference typeReference)
+		bool IsTypeInterestingForDataflow (TypeReference typeReference)
 		{
-			// We will accept any System.Type* as interesting to enable testing
-			// It's necessary to be able to implement a custom "Type" type in tests to validate
-			// the correct propagation of the annotations on "this" parameter. And tests can't really
-			// override System.Type - as it creates too many issues.
-			// It also covers TypeBuilder and RuntimeType.
 			return typeReference.MetadataType == MetadataType.String ||
-				(typeReference.Name.Contains ("Type") && typeReference.Namespace.StartsWith ("System"));
+				_hierarchyInfo.IsSystemType(typeReference) ||
+				(typeReference.Name == "IReflect" && typeReference.Namespace == "System.Reflection");
```

Comment:
Won't we need to also allow annotations on types which implement `IReflect`? Otherwise it won't be possible to implement the interface without warnings from the linker.